tests/trainer/connectors/test_signal_connector.py

```diff
@@ -37,18 +37,20 @@ def test_fault_tolerant_sig_handler(register_handler, terminate_gracefully, tmpd
         def handler(*_):
             pass
 
-        signal.signal(signal.SIGUSR1, handler)
+        signal.signal(signal.SIGTERM, handler)
 
     class TestModel(BoringModel):
         def training_step(self, batch, batch_idx):
             if terminate_gracefully or register_handler:
-                os.kill(os.getpid(), signal.SIGUSR1)
+                os.kill(os.getpid(), signal.SIGTERM)
                 sleep(0.1)
             return super().training_step(batch, batch_idx)
 
     model = TestModel()
 
-    with mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": str(int(terminate_gracefully))}):
+    with mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": str(int(terminate_gracefully))}), mock.patch(
+        "os._exit"
+    ):
```

Comment:
Do we need the exit mock still?

Reply:
yes, it is supposed to prevent an actual call to the real exit, though theoretically, even though this works, one should set the target of the patch where os._exit is used (the signal connector)

Reply:
Tested locally and seems to be passing without it. It doesn't look to be necessary after the lastest commits

Reply:
Then something is not working as intended. Based on the explanation of @tchaton, when exiting gracefully we trigger os._exit. During running the pytest, this should actually exit the test run but this is not what we want. Therefore the mock.

Reply:
I removed them !